src/storage/src/hummock/compactor/shared_buffer_compact.rs

```diff
@@ -413,11 +413,11 @@ pub async fn merge_imms_in_memory(
             // a delete range in the merged imm which it belongs to. Therefore we need
             // to construct a corresponding delete key to represent this.
             versions.push((
-                earliest_range_delete_which_can_see_key,
+                EpochWithGap::new_from_epoch(earliest_range_delete_which_can_see_key),
                 HummockValue::Delete,
             ));
         }
-        versions.push((epoch, value));
+        versions.push((EpochWithGap::new_from_epoch(epoch), value));
```

Comment:
This can generate multiple entries with the same epoch without spill offset for the same key. Will this be a problem?

For example, if we have imm1=[k1: epoch1 | offset1] and imm2=[k1: epoch1 | offset2], we will generate imm3={k1: epoch1 | 0, k1: epoch1 | 0}. I am not sure whether the ordering is still correct.

I think we need to either use the epoch with spill offset here or compact the entries with the same epoch by keeping only one entry with the largest spill offset.